js_modules/dagster-ui/packages/ui-core/src/overview/OverviewResourcesRoot.tsx

```diff
@@ -44,17 +50,28 @@ export const OverviewResourcesRoot = () => {
       notifyOnNetworkStatusChange: true,
     },
   );
-  const {data, loading} = queryResultOverview;
-  useBlockTraceOnQueryResult(queryResultOverview, 'OverviewResourcesQuery');
+  const {data, loading: queryLoading} = queryResultOverview;
+  useBlockTraceUntilTrue('OverviewResources', !!data || !workspaceLoading);
   const refreshState = useQueryRefreshAtInterval(queryResultOverview, FIFTEEN_SECONDS);
 
   // Batch up the data and bucket by repo.
   const repoBuckets = useMemo(() => {
     const visibleKeys = visibleRepoKeys(visibleRepos);
-    return buildBuckets(data).filter(({repoAddress}) =>
+    const cachedEntries = Object.values(cachedData).filter(
+      (location): location is Extract<typeof location, {__typename: 'WorkspaceLocationEntry'}> =>
+        location.__typename === 'WorkspaceLocationEntry',
+    );
+    const workspaceOrError = data?.workspaceOrError;
+    const entries =
+      workspaceOrError?.__typename === 'Workspace'
+        ? workspaceOrError.locationEntries
+        : cachedEntries;
+    return buildBuckets(entries).filter(({repoAddress}) =>
```

Comment:
I'm not sure I see the benefit here - we can fill in the cached data from the workspace context, but the user isn't going to "see" anything on this page until the `queryResultOverview` query returns with the `ResourceEntryFragment` info? 

Does this allow the user to see empty sections as opposed to no sections while the page is loading?

I think it'll make more sense when the queries on the page aren't getting data that is available on the workspace context, and I think that's what the next PR in your stack might do, but I guess I don't know that this "cached data until real data" is a useful intermediate state, it's just complicated to reason about where info is coming from and what we're waiting for.

Reply:
ResourceEntryFragment was added to the query backing the workspace context in this PR. The reason being that its a definition and isn't going to change without the location itself being updated.

Reply:
Basically the core thing we're trying to achieve with these PRs is to frontload all definitions and cache them. Then individual pages should only be querying for stateful information. In the case of this page it's all definition level information so the page doesn't need to make the query it's making but I didn't change that just yet in case something unexpected comes up.